ote_sdk/ote_sdk/utils/argument_checks.py

```diff
@@ -0,0 +1,206 @@
+    check_optional_parameters_type(optional_parameters)
+
+
+def check_nested_elements_type(iterable, parameter_name, expected_type):
+    """Function raises ValueError exception if one of elements in collection has unexpected type"""
+    for element in iterable:
+        check_parameter_type(
+            parameter=element,
+            parameter_name=f"nested {parameter_name}",
+            expected_type=expected_type,
+        )
+
+
+def check_several_lists_elements_type(parameter_name_expected_type: list):
+    """
+    Function checks if parameters lists exist and raises ValueError exception if lists elements have unexpected type
+    :param parameter_name_expected_type: list with tuples that contain parameter with nested elements, name for
+    exception message and expected type
+    """
+    for parameter, name, expected_type in parameter_name_expected_type:
+        if parameter:
+            check_nested_elements_type(
+                iterable=parameter, parameter_name=name, expected_type=expected_type
+            )
+
+
+def check_parameter_str_class_name(parameter, parameter_name, expected_class_names):
+    """Function raises ValueError exception if string class name is not equal to expected"""
+    parameter_class_name = type(parameter).__name__
+    if (parameter_class_name in expected_class_names) is False:
+        raise ValueError(
+            f"Unexpected type of '{parameter_name}' parameter, expected: {expected_class_names}, actual: "
+            f"{parameter_class_name}"
+        )
+
+
+def check_dictionary_keys_values_type(
+    parameter, parameter_name, expected_key_class, expected_value_class
+):
+    """Function raises ValueError exception if dictionary keys or values have unexpected type"""
+    for key, value in parameter.items():
+        parameter_type = type(key)
+        if not isinstance(key, expected_key_class):
+            raise ValueError(
+                f"Unexpected type of nested '{parameter_name}' dictionary key, expected: {expected_key_class}, "
+                f"actual: {parameter_type}"
+            )
+        parameter_type = type(value)
+        if not isinstance(value, expected_value_class):
+            raise ValueError(
+                f"Unexpected type of nested '{parameter_name}' dictionary value, expected: {expected_value_class}, "
+                f"actual: {parameter_type}"
+            )
+
+
+def check_several_dictionaries_keys_values_type(parameter_name_expected_type: list):
+    """
+    Function checks if parameters dictionaries exist and raises ValueError exception if their key or value have
+    unexpected type
+    :param parameter_name_expected_type: list with tuples that contain dictionary parameter, name for exception message
+    and expected type
+    """
+    for (
+        parameter,
+        name,
+        expected_key_class,
+        expected_value_class,
+    ) in parameter_name_expected_type:
+        if parameter:
```

Comment:
```suggestion
        if parameter is not None:
```

Reply:
BTW, why `if parameter` or `if parameter is not None` should be here?
I thought that the restriction should be fulfilled for all elements of the dicts.

Reply:
I added this check because the "parameter" variable is the optional and may be None-type object (ex. "model_adapters" dict-type parameter for ModelEntity class), dictionary keys and values are checked in the "check_dictionary_keys_values_type" function